src/gluonnlp/vocab/bert.py

```diff
@@ -70,25 +78,30 @@ class BERTVocab(Vocab):
         A token used to separate sentence pairs for BERT.
     cls_token : hashable object or None, default '[CLS]'
         Classification symbol for BERT.
-    reserved_tokens : list of hashable objects or None, default [mask_token, sep_token, cls_token],
-        A list of reserved tokens (excluding `unknown_token`) that will always be indexed, such as
-        special symbols representing padding, beginning of sentence, and end of sentence. It cannot
-        contain `unknown_token` or duplicate reserved tokens. Keys of `counter`, `unknown_token`,
-        and values of `reserved_tokens` must be of the same hashable type. Examples: str, int, and
-        tuple.
+    reserved_tokens : list of hashable objects or None, default None
+        A list specifying additional reserved tokens that will always be
+        indexed similar to `padding_token`, `bos_token` or `eos_token`.
+        `reserved_tokens` cannot contain `unknown_token` or duplicate reserved
+        tokens.
+        Keys of `counter`, `unknown_token`, and values of `reserved_tokens`
+        must be of the same hashable type. Examples of hashable types are str,
+        int, and tuple.
+    identifiers_to_tokens : dict mapping elements of reserved_tokens to str or None, default None
```

Comment:
I'd suggest document how excessive tokens (i.e. tokens without a counter part in reserved_tokens) are handled and what happens to reserved tokens without an entry in this mapping.

Would it be possible to merge this with the reserved_tokens argument to avoid dealing with the mapping mismatch?

Reply:
Actually `identifiers_to_tokens` can specify identifiers for any token in the vocabulary, not just reserved tokens. I will correct the API doc. Only allowing to expose "valild" tokens that are part of the vocab as attributes acts as sanity check, by explicitly raising an error if the specified token is not valid, eg. due to a typo, instead of assuming that the new token must be added.

Reply:
One downside of merging this with reserved_tokens is that it complicates the case where only some of the reserved tokens are exposed as attributes.

Reply:
The original intention of reserved_tokens is for special tokens to be registered regardless of counts. If any regular token can be exposed as attribute, I'm not sure if we need the concept of reserved tokens anymore.